src/API/useCharacters.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+
+interface Result {
+  info: {
+    count: number;
+    next: string | null;
+    pages: number;
+    prev: string | null;
+  };
+  results: Character[];
+}
+export interface Character {
+  id: number;
+  image: string;
+  name: string;
+  species: string;
+  type: string;
+  gender: "Female" | "Male" | "Genderless" | "unknown";
+  origin: { name: string; url: string };
+
+  status: "Dead" | "Alive" | "Unknown";
+  location: {
+    name: string;
+    url: string;
+  };
+  episode: string[];
+  url: string;
+  created: string;
+}
+
+const useCharacters = (
+  pageNumber: number,
+  CharacterName: string,
+  CharacterStatus: string
+) => {
+  const [fetchedCharacters, setFetchedCharacters] = useState<Result | null>(
+    null
+  );
+  const characterApiEndpoint = `https://rickandmortyapi.com/api/character/?page=${pageNumber}&name=${CharacterName}&status=${CharacterStatus}`;
+
+  useEffect(() => {
+    fetch(characterApiEndpoint)
```

Comment:
I would go for the async/await syntax, that is a bit easier to read, but again no need to change.

Reply:
Yes, sure, I can apply in future :)